aip/general/0213.md

```diff
@@ -18,24 +18,105 @@ into taking major version changes or even lead to dependency conflicts.
 
 However, there are also cases where common structures are valuable, especially
 where a concept is well-known and it is sufficiently clear that it will not
-change. Common protos serve this use case.
+change. Common protos serve this use case. There are two forms of common protos:
+organization-specific and global.
 
-## Guidance
+## Guidance for API protos
 
-The public representation of APIs **should** be self-contained, meaning that
+The public representation of APIs **must** be self-contained, meaning that
 all protos used by the API originate in the same proto `package`, except for
 common protos, which **may** be used freely in any API.
 
 APIs **must not** define a set of API-specific common protos which live outside
-of its versioning structure. This prevents independent movement of particular
-versions and also causes problems for client libraries in many languages that
-compile the proto messages into classes.
+their versioning structure. (This is sometimes attempted as a way of reusing
```

Comment:
I find the term "versioning structure" a little unclear here. It's really saying that you can't depend on common protos unless they have the same versioning scheme and version, correct?

Reply:
Not quite - as common protos don't *have* a versioning scheme. It might be simpler just to say "outside their versioned directory" or similar. We could easily given an example (ideally a fictional example, although there are concrete examples that we have today...)

How about:

> APIs **must not** define a set of API-specific common protos outside their versioned directory.
>
> *Example*: suppose the Google Library API has major versions v1 and v2. Protos specific to the Google Library API must be in `google/library/v1` and `google/library/v2`; they must not can't share protos defined in a common API-specific `google/library/common` directory, nor put protos into just `google/library`.

(I don't know if we have a common style for "example" paragraphs - the above roughly matches what we do in the C# standard...)